src/service/ComposeService.ts

```diff
@@ -293,6 +297,66 @@ export class ComposeService {
                             ...n.compose,
                         }),
                     );
+                    if (n.httpsProxy) {
+                        if (presetData.httpsProxies?.length) {
+                            if (presetData.nodes?.length && presetData.nodes[0].host) {
+                                const httpsProxyPreset = presetData.httpsProxies[0];
+                                httpsProxyPreset.excludeDockerService = false;
+                                const restGatewayDomains = resolveHttpsProxyDomains(
+                                    presetData.nodes[0].host,
+                                    `http://${n.name}:${internalPort}`,
+                                );
+                                if (httpsProxyPreset.domains) {
+                                    // domains already defined
+                                    if (httpsProxyPreset.domains.indexOf(n.name) < 0) {
+                                        // if rest-gateway not in the domains then add it
+                                        httpsProxyPreset.domains += ',' + restGatewayDomains;
+                                    }
+                                } else {
+                                    httpsProxyPreset.domains = restGatewayDomains;
+                                }
+                                if (typeof n.httpsProxy === 'number' && httpsProxyPreset.openPort === 3001) {
+                                    httpsProxyPreset.openPort = n.httpsProxy;
+                                }
+                            } else {
+                                throw new Error(`HTTPS on ${n.name} is enabled, 'host' property must be set to a valid DNS record.`);
+                            }
+                        }
+                    }
+                }),
+        );
+
+        await Promise.all(
+            (presetData.httpsProxies || [])
+                .filter((d) => !d.excludeDockerService)
+                .map(async (n) => {
+                    const internalPort = 443;
+
+                    const volumes = [vol(`../${targetGatewaysFolder}/https-proxy`, nodeWorkingDirectory, false)];
+                    services.push(
+                        await resolveService(n, {
+                            container_name: n.name,
+                            // user, // TODO fix the user permissions issue
```

Comment:
Does the user need Sudo to clean up the target folder? Are sudo files stored in the HTTP proxy volumen?

Reply:
Not I'm afraid, I think it's because of the 's6-overlay' that the [https-portal docker](https://github.com/SteveLTN/https-portal/blob/master/Dockerfile) file is using. Couldn't find a decent solution yet but will continue to research; I think that docker file needs to do the setup with root user and then use a less privileged user. We might end up extending that docker file and keeping a less privileged user.

Reply:
`We might end up extending that docker file and keeping a less privileged user.` - probably, but given lack of `user` above, I'm guessing this should be solved sooner rather than later (read: prior to merging to `main`)

Reply:
I removed the user and the volume parts from `ComposeService`. The container will hold the certificates produced unless there is an image upgrade which doesn't happen frequently, so we are safe from the produced file permissions issues. There will be another PR for the `symbol-bootstrap stop` command improvement (which will prevent containers to be removed when stopped).